NBug/Core/Reporting/BugReport.cs

```diff
@@ -49,6 +49,11 @@ internal ExecutionFlow Report(Exception exception, ExceptionThread exceptionThre
 					this.CreateReportZip(serializableException, report);
 				}
 
+                if (!(Settings.DeferredReportingOnApplicationExit && uiDialogResult.Execution == ExecutionFlow.BreakExecution)) 

```

Comment:
We can keep the name `DeferredReporting` as `DeferredReportingOnApplicationExit` is no more descriptive than the former. First timers will always need to read the XML docs, which is well explained. It also always defaults to true: https://github.com/soygul/NBug/blob/master/NBug/app.config#L48 so you can rewrite this `if` to be more explicit:

```csharp
// If NBug is configured not to delay error reporting and user did not select to exit the app immediately,
// start dispatching the bug report right away
if (!Settings.DeferredReporting && uiDialogResult.Execution == ExecutionFlow.ContinueExecution)
```